packages/techdocs-cli/cypress/integration/backstage_serve.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/// <reference types="cypress" />
+describe('TechDocs Live Preview - Backstage Serve', () => {
+  it('successfully serves documentation', () => {
+    cy.visit(`${Cypress.env('backstageBaseUrl')}/docs/default/component/local`);
+    cy.contains('hello mock docs');
+  });
+
+  it('successfully navigates to sub page of documentation', () => {
+    cy.contains('SubDocs').click();
+    cy.contains('Home 2').click();
+    cy.contains(
+      'This is an md file in another docs folder using the MkDocs Monorepo Plugin',
+    );
+  });
+
+  it('successfully renders all Backstage main elements', () => {
+    cy.contains('header', 'Live preview environment');
+    cy.get('[data-testid="sidebar-root"]')
+      .children()
+      .should('have.length.gt', 0);
+  });
+
+  it('successfully renders all extracted MkDocs main elements', () => {
+    // as it gets replaced by Backstage header
+    cy.get('.md-header').should('have.length', 0);
```

Comment:
Do you think it's a good idea not to rely on `mkdocs-material` classes and adding data-testids to these elements would be safer to prevent these tests from breaking?

Reply:
My thinking around it was - if the classes changes - we want the test to break 😬 Or? As we may rely on it in the Reader and therefore want to make sure we catch the update. But maybe worth checking both actually? What do you think @camilaibs?

Reply:
Classes are generated by `mkdocs`, right? If so, will the tests only work for a Reader implementation that uses `mkdocs`? When [this](https://github.com/backstage/backstage/pull/9385) pull request is merged and people use a different Reader implementation what would you advise them to use for testing? Maybe if they use the same test ids the tests will continue to work regardless of the Reader chosen and the CSS used by it.

Reply:
> different Reader implementation

Then the embedded app needs to be updated anyways, as its currently using the one Reader that exist. So if/when that happens we can update the tests here too!

Reply:
You're right, so we can update the tests later, I agree with you!